torch_xla/csrc/aten_xla_type.cpp

```diff
@@ -145,6 +146,67 @@ void AtenInitialize() {
 
 }  // namespace
 
+namespace xla_pre_autograd {
+
+struct MaxPool2dAutogradFunction
+    : public torch::autograd::Function<MaxPool2dAutogradFunction> {
+  static torch::Tensor forward(torch::autograd::AutogradContext* ctx,
+                               torch::Tensor self,
+                               torch::IntArrayRef kernel_size,
+                               torch::IntArrayRef stride,
+                               torch::IntArrayRef padding,
+                               torch::IntArrayRef dilation, bool ceil_mode) {
+    ctx->saved_data["kernel_size"] = kernel_size;
+    ctx->saved_data["stride"] = stride;
+    ctx->saved_data["padding"] = padding;
+    ctx->saved_data["dilation"] = dilation;
+    ctx->saved_data["ceil_mode"] = ceil_mode;
+    // Lowering when ceil_mode or dilation is set not supported yet.
+    if (IsNonTrivialDilation(dilation)) {
+      auto results = AtenXlaTypeDefault::max_pool2d_with_indices(
+          self, kernel_size, stride, padding, dilation, ceil_mode);
+      ctx->save_for_backward({self, std::get<1>(results)});
+      return std::get<0>(results);
+    }
+    ctx->save_for_backward({self});
+    auto outputs = XLATensor::max_pool_nd(
+        bridge::GetXlaTensor(self), /*spatial_dim_count=*/2,
+        XlaHelpers::I64List(kernel_size), XlaHelpers::I64List(stride),
+        XlaHelpers::I64List(padding), ceil_mode);
+    return bridge::AtenFromXlaTensor(std::get<0>(outputs));
+  }
+
+  static torch::autograd::variable_list backward(
+      torch::autograd::AutogradContext* ctx,
+      torch::autograd::variable_list grad_output) {
+    auto kernel_size = ctx->saved_data["kernel_size"].toIntList().vec();
+    auto stride = ctx->saved_data["stride"].toIntList().vec();
```

Comment:
We do not need to turn those into vectors and make allocations, do we?

Reply:
`toIntList()` returns `c10::List` but we expect `ArrayRef` later. There's no constructor `ArrayRef(c10::List)` available so I converted it to vector.